dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/datavalueset/DefaultDataValueSetService.java

```diff
@@ -1155,13 +1162,47 @@ else if ( dataValueSet.getAttributeCategoryOptions() != null && dataSet != null
                 trimToNull( dataValue.getCategoryOptionCombo() ),
                 context.getCategoryOptionComboCallable().setId( trimToNull( dataValue.getCategoryOptionCombo() ) ) ) )
             .attrOptionCombo(
-                dataSetContext.getOuterAttrOptionCombo() != null ? dataSetContext.getOuterAttrOptionCombo()
-                    : context.getOptionComboMap().get( trimToNull( dataValue.getAttributeOptionCombo() ),
-                        context.getAttributeOptionComboCallable()
-                            .setId( trimToNull( dataValue.getAttributeOptionCombo() ) ) ) )
+                findAttributeOptionCombo( dataSetContext, dataValue, context ) )
             .build();
     }
 
+    private CategoryOptionCombo findAttributeOptionCombo( DataSetContext dataSetContext,
```

Comment:
We have some code for this in `InputUtils#getAttributeOptionCombo`, could you check if we can harmonize?

Reply:
this change has been removed from this PR